tests/devices/test_default_qubit.py

```diff
@@ -1956,6 +1959,27 @@ def test_apply_two_qubit_op_reverse(self, op, method, inverse):
         state_out_einsum = np.einsum("abcd,idc->iba", matrix, self.state)
         assert np.allclose(state_out, state_out_einsum)
 
+    @pytest.mark.parametrize("op, method", three_qubit_ops)
+    def test_apply_three_qubit_op(self, op, method, inverse):
+        """Test if the application of three qubit operations is correct."""
+        state_out = method(self.state, axes=[0, 1, 2])
+        op = op(wires=[0, 1, 2])
+        matrix = op.inv().matrix if inverse else op.matrix
+        matrix = matrix.reshape((2, 2, 2, 2, 2, 2))
+        state_out_einsum = np.einsum("abcdef,def->abc", matrix, self.state)
+        assert np.allclose(state_out, state_out_einsum)
+
+    @pytest.mark.parametrize("op, method", three_qubit_ops)
+    def test_apply_three_qubit_op_reverse(self, op, method, inverse):
```

Comment:
It could be nice to add a third case too, where out of the two control wires one is greater and one is smaller than the target wire (assuming the wires are represented with consecutive integers), e.g., testing the `axes=[2, 0, 1]` case.